Updater/save healthcheck

Hoxmot opened the pull request:
I've added saving health check timestamps to the database and prepared handlers for unhealthy targets.

todos:
- [x] I have to attach my code to sending emails in #10 
- [x] Somethings wrong with taking out address from the database. It's getting cached :/
- [ ] Some tests would be nice I guess

Hoxmot committed:
[updater] saving healthchecks to database

Hoxmot committed:
[updater] updating targets

* Adding information about last_healthy timestamp for healthy targets.
* Checking if we should send email for unhealthy targets
* Saving timestamp of email sent for first admin
* Saving information on sending email for second admin

js361014 commented on updater/src/src/main/java/service/health/check/updater/managers/AddressManager.java:
```diff
@@ -0,0 +1,67 @@
+        select.select(root);
+        select.where(cb.equal(root.get(Address_.host), host),
+                     cb.equal(root.get(Address_.port), port));
+        return entityManager.createQuery(select).getSingleResult();
```
```
Address address = entityManager.createQuery(select).getSingleResult();
Session session = (Session) entityManager.getDelegate();
session.refresh(address);
 return address;
```

js361014 commented on updater/src/src/main/java/service/health/check/updater/CheckedAddressHandlerUtil.java:
```diff
@@ -0,0 +1,38 @@
+    private static final long SECONDS_TO_MILLIS = 1000;
+
+    public static boolean shouldSendMailToFirstAdmin(Address address, Timestamp now) {
+        if (address.getLastHealthy() == null) {
```
Dlaczego tak?

Hoxmot replied:
Potrzebuję więcej szczegółów :P 
jeżeli chodzi o to, czemu nie chcę wysyłać maila jak nie ma health checka, to dlatego, że tak – w sensie, uznałem, że jeżeli dodaliśmy nowy cel i jeszcze ani razu nie udało nam się dostukać, to znaczy, że jeszcze setupujemy tamten cel i dlatego nie jest OK

js361014 commented on database-models/src/src/main/java/service/health/check/models/Address.java:
```diff
@@ -28,9 +30,27 @@
 	@Type(type= "org.hibernate.type.BooleanType")
-	@Column(name = "healthy", nullable = false)
-	private Boolean healthy;
+	@Column(name = "second_notification_sent")
```
nullable = false

Hoxmot replied:
zmieniłem jak to działa i to już nie jest Boolean, a Timestamp

Hoxmot committed:
[updater] problem with cached address fixed

Fixed a problem with address taken from the database being cached in some way,
so that it didn't contain the changes made, which cased the email sending algo
to malfunction

Hoxmot committed:
Merge branch 'master' of github.com:js361014/serviceHealthCheck into updater/save-healthcheck

Hoxmot committed:
[uodater] sending emails for dead targets

Hoxmot committed:
Updater added to GKE config

Hoxmot committed:
[front] duty confirm

Hoxmot committed:
Configured updater and frontend for GKE

* Configured updater deployment for GKE
* Configured fronted service&deployment for GKE and handling externat traffic
* Added description on manual E2E testing on GKE

Hoxmot committed:
Added mail config for GKE

Added mailiing server config for sending notifications

Hoxmot committed:
mail.yaml fix
